client/widmgr/mock.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package widmgr
+
+import (
+	"crypto/ed25519"
+	"fmt"
+	"slices"
+	"time"
+
+	"github.com/go-jose/go-jose/v3"
+	"github.com/go-jose/go-jose/v3/jwt"
+
+	"github.com/hashicorp/nomad/helper/uuid"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// MockWIDMgr allows TaskRunner unit tests to avoid having to setup a Server,
+// Client, and Allocation.
+type MockWIDMgr struct {
```

Comment:
```suggestion
type MockWIDSigner struct {
```

Alternatively we could embed a `*WIDMgr` here and have the mock work as both entities, but that could be a bit confusing?

Reply:
Renamed in [e3dd3d3](https://github.com/hashicorp/nomad/pull/18431/commits/e3dd3d3f15eda095a73e2d73cc7fa349358808c5)